### PR TITLE
Example Logging Presets

### PR DESCRIPTION
**Pull request by nniehoff**
The idea behind this PR is to provide an easy solution to quickly setup some basic logging configurations without having to research and understand the entire structure of the `LOGGING` dictionary.  The idea would be to allow the user to specify an entire LOGGING config via an env variable (see PR #197) or if they don't want to have some predefined `NAUTOBOT_LOGGING_PRESET` values which set it up for them which would be read from the environment.  The example here provides 3 `console` (useful for docker), `files` (useful for some VM deployments), and `journal` which uses the systemd journal to collect logs.  The journal is a newer way to collect logs and requires some additional packages which also need to be built from source so it might not be an example we want to go with but I wanted to put this idea out there.

### CODE REVIEW

**Comment by jvanderaa** [comment]
I think this is terrific. Having these preset and available is perfect for an enterprise app.

File: nautobot/core/templates/nautobot_config.py.j2 [hunk 1]
```diff
@@ -168,7 +168,104 @@ INTERNAL_IPS = ("127.0.0.1", "::1")
...
+            "format": "%(asctime)s.%(msecs)03d %(levelname)-7s %(name)-20s %(filename)-15s %(funcName)30s() :\n  %(message)s",
+            "datefmt": "%H:%M:%S",
+        },
+    },
+    "handlers": {},
+    "loggers": {},
+    "root": {
+        "level": LOG_LEVEL,
+    },
+}
```
Comment (jathanism) [inline nautobot/core/templates/nautobot_config.py.j2:193]: This is a great default but it's apparently getting clobbered by the `if...else` below.

File: nautobot/core/templates/nautobot_config.py.j2 [hunk 2]
```diff
@@ -168,7 +168,104 @@ INTERNAL_IPS = ("127.0.0.1", "::1")
...
+        "level": "DEBUG",
+        "class": "systemd.journal.JournaldLogHandler",
+        "formatter": "verbose" if DEBUG else "normal",
+    }
+    LOGGING["loggers"] = {
+        "django": {"handlers": ["systemd"], "level": DJANGO_LOG_LEVEL},
+    }
+    LOGGING["root"]["handlers"] = ["systemd"]
+else:
+    LOGGING = {}
```
Comment (jathanism) [inline nautobot/core/templates/nautobot_config.py.j2:268]: IMO just ditch the logic here and define all of these handlers by default. Then just replace the logic with a bit of user-facing documentation, e.g. "If you want to log to a file use this handler, or log to console use this one. If you want to change the path to the log file in the default `nautobot_log` handler, change the `filename` variable."

For example, imagine something in the docs like "To enable debug logging to using the `nautobot_log` handler and change the file location in your `nautobot_config.py`":

```
LOGGING["loggers]["django"] = {"handlers": ["nautobot_log"], level="DEBUG"}
LOGGING["handlers"]["nautobot_log"]["filename"] = "/path/to/nautobot.log"

**Review by jathanism — CHANGES_REQUESTED** [review]

File: nautobot/core/templates/nautobot_config.py.j2 (hunk 2 above)
Comment (nniehoff) [reply nautobot/core/templates/nautobot_config.py.j2]: Ok I removed the logic for simply a preset to configure logging for you and moved it all to the docs.  I set the basic configuration in `settings.py` so if people want to override the whole `LOGGING` config they can do that easily in the config

**Pull request closed without merging**